From 6a73d9ddf7d14819148502cec0b742ebae8e709f Mon Sep 17 00:00:00 2001
From: "Ch1llb0/skin.osmc" <36623079+Ch1llb0@users.noreply.github.com>
Date: Fri, 1 Mar 2019 18:48:08 +0100
Subject: [PATCH] Video info dialog fix

DialogVideoInfo.xml:
- add fadelabel for file path and name when refresh button is focused
- add director button

addon.xml:
- update changelog

Changelog.md:
- update changelog
---
 16x9/DialogVideoInfo.xml | 19 +++++++++++++++++++
 Changelog.md             |  8 ++++++--
 addon.xml                |  2 +-
 3 files changed, 26 insertions(+), 3 deletions(-)

diff --git a/16x9/DialogVideoInfo.xml b/16x9/DialogVideoInfo.xml
index 25f6b8b..769b35f 100755
--- a/16x9/DialogVideoInfo.xml
+++ b/16x9/DialogVideoInfo.xml
@@ -47,6 +47,19 @@
 					<aspectratio aligny="center">keep</aspectratio>
 				</control>
 			</control>
+			
+			<control type="fadelabel">
+				<left>150</left>
+				<bottom>180</bottom>
+				<width>1620</width>
+				<height>36</height>
+				<font>Font27</font>
+				<align>center</align>
+				<label>$INFO[ListItem.FileNameAndPath]</label>
+				<textcolor>$VAR[TextColor2]</textcolor>
+				<pauseatend>5000</pauseatend>
+				<visible>Control.HasFocus(6)</visible>
+			</control>
 
 			<!-- Info -->
 			<control type="group">
@@ -439,6 +452,12 @@
 					<label>206</label>
 					<onclick condition="String.IsEqual(Control.GetLabel(5),$LOCALIZE[207])">SetFocus(50)</onclick>
 				</control>
+				<!-- Director Button -->
+				<control type="button" id="13">
+					<width>Auto</width>
+					<label>20339</label>
+					<visible>!String.IsEmpty(ListItem.Director)</visible>
+				</control>
 				<!-- Refresh Button -->
 				<control type="button" id="6">
 					<width>Auto</width>
diff --git a/Changelog.md b/Changelog.md
index 0f35f4a..95d14f8 100644
--- a/Changelog.md
+++ b/Changelog.md
@@ -9,7 +9,8 @@ _New_
 - new color options (color sets, background gradients, adjustable opacity)
 - add PVR channel number input dialog
 - add PVR timeshift status dialog
-- ddd welcome dialog on non-OSMC devices
+- add welcome dialog on non-OSMC devices
+- add director button to video info dialog
 
 _Improved_
 - adjust syntax, values, labels and infobools to match v18 requirements
@@ -18,6 +19,7 @@ _Improved_
 - streamline OSD animations
 - add missing adjustable plot fonts
 - let favourites dialog behave like a normal window
+- add file path and name to refresh button in video info dialog
 
 _Fixed_
 - show proper game widget title when not using skinshortcuts script
@@ -68,7 +70,7 @@ Addon.Browser.xml:
 Custom_Cache_Progress.xml:
 - replace colors by new color variables
 
-Custom_Welcom.xml
+Custom_Welcome.xml
 - add new Welcome dialog for non-OSMC devices
 
 Defaults.xml:
@@ -208,6 +210,8 @@ DialogSubtitles.xml:
 
 DialogVideoInfo.xml:
 - replace colors by new color variables
+- add director button
+- add file path and name to refresh button
 
 EventLog.xml:
 - replace colors by new color variables
diff --git a/addon.xml b/addon.xml
index 510e53f..5d12406 100644
--- a/addon.xml
+++ b/addon.xml
@@ -14,7 +14,7 @@
 		<platform>all</platform>
 		<summary lang="en">The default skin for OSMC</summary>
 		<description lang="en">Original skin: Andy Morton[CR]Original design: Simon Brunton[CR]Skinner: Julian Michel</description>
-		<news>OSMC Kodi v18 update[CR][CR][B]New[/B][CR]- add new v18 subtitle settings OSD during fullscreen video playback[CR]- add new games section to match v18 requirements[CR]- add games section[CR]- add new resolution select button/dialog in video player[CR]- add player icon to now playing dialog[CR]- add new dependency button in addon info dialog to match v18 requirements[CR]- new color options (color sets, background gradients, adjustable opacity)[CR]- add PVR channel number input dialog[CR]- add PVR timeshift status dialog[CR]- add welcome dialog on non-OSMC devices[CR][CR][B]Improved[/B][CR]- adjust syntax, values, labels and infobools to match v18 requirements[CR]- adjust PVR section to match v18 requirements[CR]- highlighting color now adjusts according to text color[CR]- streamline OSD animations[CR]- add missing adjustable plot fonts[CR]- let favourites dialog behave like a normal window[CR][CR][B]Fixed[/B][CR]- show proper game widget title when not using skinshortcuts script[CR]- highlighting is now more consistent[CR]- fix current position/time remaining and current time/end time for PVR playback[CR]- hide deprecated previous/next channel buttons in PVR playback OSD[CR]- fix background of subtitle settings window[CR]- fix layout of PVR playback dialogs</news>
+		<news>OSMC Kodi v18 update[CR][CR][B]New[/B][CR]- add new v18 subtitle settings OSD during fullscreen video playback[CR]- add new games section to match v18 requirements[CR]- add games section[CR]- add new resolution select button/dialog in video player[CR]- add player icon to now playing dialog[CR]- add new dependency button in addon info dialog to match v18 requirements[CR]- new color options (color sets, background gradients, adjustable opacity)[CR]- add PVR channel number input dialog[CR]- add PVR timeshift status dialog[CR]- add welcome dialog on non-OSMC devices[CR]- add director button to video info dialog[CR][CR][B]Improved[/B][CR]- adjust syntax, values, labels and infobools to match v18 requirements[CR]- adjust PVR section to match v18 requirements[CR]- highlighting color now adjusts according to text color[CR]- streamline OSD animations[CR]- add missing adjustable plot fonts[CR]- let favourites dialog behave like a normal window[CR]- add file path and name to refresh button in video info dialog[CR][CR][B]Fixed[/B][CR]- show proper game widget title when not using skinshortcuts script[CR]- highlighting is now more consistent[CR]- fix current position/time remaining and current time/end time for PVR playback[CR]- hide deprecated previous/next channel buttons in PVR playback OSD[CR]- fix background of subtitle settings window[CR]- fix layout of PVR playback dialogs</news>
 	</extension>
 	<assets>
 		<icon>icon.png</icon>
